File: test_ws.py
"""
Smoke test for the Revolut Gateway WebSocket server.

Usage:
    uv run python test_ws.py
"""

from __future__ import annotations
import asyncio
import json
import uuid

import websockets


WS_URL = "ws://localhost:18800/"


async def main() -> None:
    print(f"Connecting to {WS_URL}...")
    async with websockets.connect(WS_URL) as ws:
        # 1. Handshake
        req_id = uuid.uuid4().hex[:8]
        await ws.send(json.dumps({
            "type": "req",
            "id": req_id,
            "method": "connect",
            "params": {
                "client": {"id": "test-client", "version": "0.1.0", "platform": "python"},
            },
        }))
        hello = json.loads(await ws.recv())
        print("hello-ok:", json.dumps(hello, indent=2))

        # 2. Send a chat message
        chat_id = uuid.uuid4().hex[:8]
        await ws.send(json.dumps({
            "type": "req",
            "id": chat_id,
            "method": "chat.send",
            "params": {"key": "test-sess", "message": "Hello from Python test!"},
        }))

        # Read until idle
        while True:
            msg = json.loads(await asyncio.wait_for(ws.recv(), timeout=120))
            print(json.dumps(msg))
            if msg.get("type") == "event" and msg.get("event") == "chat.idle":
                break

    print("Done.")


if __name__ == "__main__":
    asyncio.run(main())

File: test-ws.js
import { WebSocket } from 'ws';
const ws = new WebSocket('ws://localhost:18800');

ws.on('open', () => {
  ws.send(JSON.stringify({
    type: 'req', id: '1', method: 'connect', params: { client: { id: 'test', version: '1', platform: 'node' } }
  }));
});

ws.on('message', (data) => {
  console.log('RECV:', data.toString());
  const msg = JSON.parse(data.toString());
  if (msg.type === 'res' && msg.id === '1') {
    ws.send(JSON.stringify({
      type: 'req', id: '2', method: 'chat.send', params: { sessionKey: 'test-sess', message: 'Hello' }
    }));
  }
});

ws.on('close', () => console.log('Closed'));
ws.on('error', (err) => console.error('Error:', err));
